src/Integration.Vsix/SonarLintDaemon/SonarLintDaemon.cs

```diff
@@ -0,0 +1,150 @@
+
+        public void Dispose()
+        {
+            if (IsRunning())
+            {
+                Stop();
+            }
+        }
+
+        public void Install()
+        {
+            Download();
+            Unzip();
+        }
+
+        public bool IsRunning()
+        {
+            if (process == null)
+            {
+                return false;
+            }
+            return !process.HasExited;
+        }
+
+        public void Start()
+        {
+            if (IsRunning())
+            {
+                throw new InvalidOperationException("Process already running");
+            }
+            if (!IsInstalled())
+            {
+                throw new InvalidOperationException("Daemon is not installed");
+            }
+
+            if (process != null)
+            {
+                process.Close();
+            }
+
+            port = TcpUtil.FindFreePort(8050);
+            process = new Process();
+            process.StartInfo.UseShellExecute = false;
+            process.StartInfo.FileName = GetExePath();
+            process.StartInfo.Arguments = GetCmdArgs(port);
+            process.StartInfo.CreateNoWindow = true;
+            process.Start();
+        }
+
+        public void Stop()
+        {
+            if (!IsRunning())
+            {
+                return;
+                // throw exception?
+            }
+            process.Kill();
+            process.WaitForExit();
+        }
+
+        public int Port()
+        {
+            return port;
+        }
+
+        public bool IsInstalled()
+        {
+            return Directory.Exists(GetInstallationPath()) && File.Exists(GetExePath());
+        }
+
+        private void Download()
+        {
+            string uri = string.Format(uriFormat, version);
+            using (var client = new WebClient())
+            {
+                client.DownloadFile(uri, GetZipFilePath());
+            }
+        }
+
+        private void Unzip()
+        {
+            if (Directory.Exists(GetInstallationPath()))
+            {
+                Directory.Delete(GetInstallationPath(), true);
+            }
+            ZipFile.ExtractToDirectory(GetZipFilePath(), storagePath);
+        }
+
+        private string GetInstallationPath()
+        {
+            return Path.Combine(storagePath, $"sonarlint-daemon-{version}-windows");
+        }
+
+        private string GetZipFilePath()
+        {
+            return Path.Combine(tmpPath, $"sonarlint-daemon-{version}-windows.zip");
+        }
+
+        private string GetExePath()
+        {
+            return Path.Combine(GetInstallationPath(), "jre", "bin", "java.exe");
+        }
+
+        private string GetCmdArgs(int port)
+        {
+            string jarPath = Path.Combine(GetInstallationPath(), "lib", $"sonarlint-daemon-{version}.jar");
+            string logPath = Path.Combine(GetInstallationPath(), "conf", "logback.xml");
+            string className = "org.sonarlint.daemon.Daemon";
```

Comment:
you should probably follow the naming convention for daemons where the name always ends with "d" :)